src/app/public/modules/auth-http-client/no-auth-interceptor.ts

```diff
@@ -0,0 +1,55 @@
+import {
+  Injectable
+} from '@angular/core';
+
+import {
+  HttpEvent,
+  HttpHandler,
+  HttpInterceptor,
+  HttpRequest
+} from '@angular/common/http';
+
+import {
+  SkyAppConfig
+} from '@skyux/config';
+
+import {
+  Observable
+} from 'rxjs/Observable';
+
+import {
+  SKY_AUTH_PARAM_AUTH
+} from './auth-interceptor-params';
+
+import {
+  BBAuthClientFactory
+} from '@skyux/auth-client-factory';
+
+import 'rxjs/add/observable/fromPromise';
+
+import 'rxjs/add/operator/switchMap';
+
+@Injectable()
+export class NoAuthInterceptor implements HttpInterceptor {
+
+  constructor(private config: SkyAppConfig) {}
+
+  public intercept(request: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> {
+    const noAuth = request.params &&
+      (!request.params.get(SKY_AUTH_PARAM_AUTH) ||
+        request.params.get(SKY_AUTH_PARAM_AUTH) === 'false');
+
+    if (noAuth) {
+      return Observable
+        .fromPromise(BBAuthClientFactory.BBAuth.getUrl(request.url))
+        .switchMap((url) => {
+          let newRequest = request.clone({
```

Comment:
This should be `const` instead of `let` since its value doesn't change after assignment. I would think TSLint would catch this.

Reply:
I'll update to a const.